### ISSUE TITLE
can this be package be DL framework agnostic?

### ISSUE THREAD

**Issue by CarloLucibello** [JuliaGenAI/FlashAttentionWrapper.jl#2]
Can this package be made to support both Flux and Lux?
It could use extensions if needed.
At the very least it could depend only LuxCore instead of the full Lux as it does right now.

**Comment by findmyway** [comment]
> Can this package be made to support both Flux and Lux?

Yes, I think Flux is supported out of the box. I'll give it a try and add an example usage in README.

> At the very least it could depend only LuxCore instead of the full Lux as it does right now.

Ah, yes. `Lux.jl` dependency was added for testing only. I'll remove it from dependencies.

> It could use extensions if needed.

I was to add `LuxCore` as an extension. But then I found that in Lux.jl, the `Chain` has a hard constraint.

https://github.com/LuxDL/Lux.jl/blob/bcbcbce26b62f009ce22f567472f2ccee83af471/src/layers/containers.jl#L469-L473

So I have to turn the `Attention` struct into a subtype of `AbstractLuxLayer`. Although in the [doc](https://lux.csail.mit.edu/stable/manual/interface), it says this is optional.

>  ... it should **optionally** subtype `Lux.AbstractLuxLayer`
